test/integration/standard_filter_test.rb

```diff
@@ -928,6 +928,47 @@ def test_where_no_target_value
     assert_equal([{ "foo" => true }, { "foo" => "for sure" }], @filters.where(input, "foo"))
   end
 
+  def test_sum
+    input = [1, 2]
+    assert_equal(3, @filters.sum(input))
+  end
+
+  def test_sum_with_numeric_strings
+    input = [1, "2"]
+    assert_equal(3, @filters.sum(input))
+  end
+
+  def test_sum_with_non_numeric_values
+    input = [3, "foo", { "foo" => "bar" }, true, nil]
+    assert_equal(3, @filters.sum(input))
+  end
+
+  def test_sum_with_property_on_indexable_map_values
+    input = [
+      { "quantity" => 1 },
+      { "quantity" => 2 },
+    ]
+
+    assert_equal(3, @filters.sum(input, "quantity"))
+    assert_equal(0, @filters.sum(input, "weight"))
+  end
+
+  def test_sum_with_property_on_indexable_non_map_values
+    input = [1, "foo", { "quantity" => 2 }]
+
+    assert_raises(Liquid::ArgumentError, "cannot select the property 'quantity'") do
+      @filters.sum(input, "quantity")
+    end
+  end
+
+  def test_sum_with_property_on_unindexable_values
+    input = [true, nil, :foo]
+    assert_equal(0, @filters.sum(input, "quantity"))
+
+    mixed_input = [true, nil, :foo, { "quantity" => 3 }]
+    assert_equal(3, @filters.sum(mixed_input, "quantity"))
+  end
```

Comment:
Implementation looks good to me! Before we merge, could we add some tests for the `#to_number` case as well as the `TypeError` case? For the later, an array of arrays should work, eg

```ruby 
assert_equal(0, @filters.sum([[1]], "quantity")) # => 0
```

Reply:
`test_sum_with_property_on_indexable_non_map_values` was intended to test the `TypeError` case
https://github.com/Shopify/liquid/blob/81fdaf81580b81e86a93256d983358185fd9f0c0/test/integration/standard_filter_test.rb#L956-L962

`test_sum_with_non_numeric_values` was intended to test the `to_number` case. 
https://github.com/Shopify/liquid/blob/81fdaf81580b81e86a93256d983358185fd9f0c0/test/integration/standard_filter_test.rb#L941-L944

But I can make these more complete – I can think of some additional edge cases to test.

Reply:
Pushed up an updated test suite that I believe covers the relevant edge cases.